src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -821,147 +827,187 @@ public unsafe void CopyTo(Array array, int index)
+                    if (BitConverter.IsLittleEndian)
+                    {
+                        bitMask = Vector512.Create(0x80402010_08040201).AsByte();
+                    }
+                    else
+                    {
+                        bitMask = Vector512.Create(0x01020408_10204080).AsByte();
+                    }
 
-                    fixed (bool* destination = &boolArray[index])
+                    for (; i < (uint)thisLength - ((uint)Vector512<byte>.Count - 1u); i += (uint)Vector512<byte>.Count)
                     {
-                        for (; (i + Vector512<byte>.Count) <= (uint)m_length; i += (uint)Vector512<byte>.Count)
+                        Vector512<long> bits = Vector512.Create(Unsafe.ReadUnaligned<long>(ref Unsafe.AddByteOffset(ref Unsafe.As<int, byte>(ref thisRef), i >> BitShiftPerByte)));
+
+                        Vector512<byte> shuffled;
+                        // Feed the shuffle indices directly to Vector512.Shuffle() for the best codegen.
+                        // See dotnet/runtime#115078
+                        if (BitConverter.IsLittleEndian)
                         {
-                            ulong bits = (ulong)(uint)m_array[i / (uint)BitsPerInt32] + ((ulong)m_array[(i / (uint)BitsPerInt32) + 1] << BitsPerInt32);
-                            Vector512<ulong> scalar = Vector512.Create(bits);
-                            Vector512<byte> shuffled = Avx512BW.Shuffle(scalar.AsByte(), shuffleMask);
-                            Vector512<byte> extracted = Avx512F.And(shuffled, bitMask);
-
-                            // The extracted bits can be anywhere between 0 and 255, so we normalise the value to either 0 or 1
-                            // to ensure compatibility with "C# bool" (0 for false, 1 for true, rest undefined)
-                            Vector512<byte> normalized = Avx512BW.Min(extracted, ones);
-                            Avx512F.Store((byte*)destination + i, normalized);
+                            shuffled = Vector512.Shuffle(bits.AsByte(),
+                                // The shuffle indices should be chosen so that with AVX512, vpshufb can be emitted instead of the slower, less avaliable(AVX512_VBMI) vpermb.
+                                Vector512.Create(0x00000000_00000000, 0x09090909_09090909, 0x12121212_12121212, 0x1B1B1B1B_1B1B1B1B,
+                                                 0x24242424_24242424, 0x2D2D2D2D_2D2D2D2D, 0x36363636_36363636, 0x3F3F3F3F_3F3F3F3F).AsByte());
                         }
+                        else
+                        {
+                            shuffled = Vector512.Shuffle(bits.AsByte(),
+                                Vector512.Create(0x03030303_03030303, 0x0A0A0A0A_0A0A0A0A, 0x11111111_11111111, 0x18181818_18181818,
+                                                 0x27272727_27272727, 0x2E2E2E2E_2E2E2E2E, 0x35353535_35353535, 0x3C3C3C3C_3C3C3C3C).AsByte());
+                        }
```

Comment:
I don't think there's any big endian platform that's supported and also supports vectors, I'd recommend just guarding the SIMD paths on LE here.

Reply:
I have already wrote these lines, so why not keep them

Reply:
> I have already wrote these lines, so why not keep them

Cause having dead, untested code both complicates the algorithm and can cause unexpected bugs if those paths get enabled later.